docs/guides/how_to_guides/creating_and_editing_expectations/how_to_create_an_expectation_suite_with_the_user_configurable_profiler.rst

```diff
@@ -242,9 +242,369 @@ This guide will help you create a new Expectation Suite by profiling your data w
+
+                - branch: # branch is optional if/then syntax. You can also just put the expectation
+                    if: $my_dateformat.success_ratio >= 0.8  # if evaluates to true in python, i.e. "" is FALSE, "%Y" is TRUE
+                                                                   # This success_ratio is provided via `details` of a parameter
+                                                                   # It is matched with the correct domain
+                                                                   # QUESTION: I removed the `parameter` in between the
+                                                                   # parameter_name and success_ratio - perhaps we can
+                                                                   # simplify these lookups - ANSWER - Cool, not sure exaclty how to implement but let's try it.
+                    then:
+                      # ExpectationConfigurationBuilder configuration goes here
+                      - name: my_expectation_configuration_builder_1
+                        class_name: DefaultExpectationConfigurationBuilder # Optional, will use Default if not supplied, otherwise will use supplied
+                        module_name: tbd # Optional, for all other
+                        expectation_type: expect_column_values_to_match_strftime_format # QUESTION: Should we change this from expectation to expectation_type?
+                        column: $my_domain_builder.domain_kwargs.column  # is this obvious/inferrable?
+                                                              # AJB - I think some may wish to be more explicit but yes
+                                                              # this should be inferrable when there is a 1-1 between
+                                                              # ProfilerRule and Domain Builder. So make this line optional.
+                        strftime_format: $my_dateformat.date_format_string # This is an expectation kwarg
+                                                                                     # (Parameters in our docs)
+                                                                                     # QUESTION: removed `parameter`
+                - branch:
+                    if: $domain.column_type in my_dateformat.DATETIME_TYPES_CONST  # Note that domain returned "column_type" in addition to "domain_kwargs" - Note also that DATEITME_TYPES_CONST doesn't yet exist but would be a class variable of the SimpleDateFormatStringParameterBuilder
+                                                                                   # QUESTION: Is this a "subdomain"? It kind of feels against the 1 domain per ProfilerRule, but makes sense. Suppose I want to reference this elsewhere, I guess for now we copy paste.
+                    then:
+                      - expectation: expect_column_values_to_be_in_type_list
```

Comment:
For consistency, we should pick one of expectation or expectation type and go with this choice in both locations - this is confusing

Reply:
Good point - going with `expectation_type` for future configs since this is what is used in ExpectationConfiguration

Reply:
See here for proposed change to "single batch Alice" https://github.com/great-expectations/great_expectations/pull/2745/files